lib/clog.rb

```diff
@@ -25,4 +25,9 @@ def self.emit(message)
     puts JSON.generate(out)
     nil
   end
+
+  def self.fail(message, &block)
```

Comment:
I'd suggest making a custom error type if we're going to start to have an abstraction here. It may not do anything interesting right now, though, e.g.

```ruby
class Clog
  Fail = Class.new(RuntimeError)
  # [...]
end
```

Also I think the `fail` method is slightly preferred when there's no possibility of re-raising.

Reply:
Initially, I used `fail`, but it calls `Clog.fail` instead of `Kernel.fail`, so infinite loop. I can explicitly use `Kernel.fail`.

Reply:
generally shadowing is bad, but I think in this case, it makes sense :)

Reply:
What do you about passing block result to `Clog.Fail` exception?